src/marten/db/query/sql/raw_statement_node.cr

```diff
@@ -0,0 +1,99 @@
+module Marten
+  module DB
+    module Query
+      module SQL
+        class RawStatementNode < PredicateNode
+          getter statement
+          getter params
+
+          def initialize(
+            @statement : String,
+            @params = [] of ::DB::Any,
+            @children = [] of PredicateNode,
+            @connector = SQL::PredicateConnector::AND, @negated = false, *args
+          )
+            @predicates = [] of Predicate::Base
+            @predicates.concat(args.to_a)
+          end
+
+          def initialize(
+            @statement : String,
+            @params : Array(::DB::Any) | Hash(String, ::DB::Any),
+            @children : Array(PredicateNode),
+            @connector : PredicateConnector,
+            @negated : Bool,
+            @predicates : Array(Predicate::Base)
+          )
+          end
+
+          def ==(other : RawStatementNode)
+            (
+              (other.statement == statement) &&
+                (other.params == params) &&
+                (other.children == children) &&
+                (other.connector == connector) &&
+                (other.negated == negated)
+            )
+          end
+
+          def clone
+            RawStatementNode.new(
+              statement: @statement,
+              params: params,
+              children: @children.map(&.clone),
+              connector: @connector,
+              negated: @negated,
+              predicates: @predicates.dup
+            )
+          end
+
+          def to_sql(connection : Connection::Base)
+            # Escape % characters to be not be considered
+            # during formatting process
+            @statement = @statement.gsub("%", "%%")
+
+            case params
+            when Array(::DB::Any)
+              sanitize_positional_parameters(connection)
+            else
+              sanitize_named_parameters(connection)
+            end
```

Comment:
This logic and the underlying methods and constants are duplicated from the `Marten::DB::Query::SQL::RawQuery` abstraction. What do you think about defining this logic into a shared `Marten::DB::Query::SQL::HasSanitizer` concern module?